types/next-auth/client.d.ts

```diff
@@ -0,0 +1,102 @@
+import { FC } from 'react';
+
+interface Session {
+    user: {
+        name: string;
+        email: string;
+        image: string;
+    };
+    accessToken: string;
+    expires: string;
+}
+
+interface GetProvidersResponse {
+    [provider: string]: SessionProvider;
+}
+
+interface SessionProvider {
+    id: string;
+    name: string;
+    type: string;
+    signinUrl: string;
+    callbackUrl: string;
+}
+
+interface GenericObject {
+    [key: string]: any;
+}
+
+interface ContextProviderProps {
+    session: Session;
+    options?: ContextProviderOptions;
+}
+
+interface ContextProviderOptions {
+    site?: string;
+    basePath?: string;
+    clientMaxAge?: number;
+}
+
+interface SetOptionsParams {
+    baseUrl?: string;
+    basePath?: string;
+    clientMaxAge?: string;
+    keepAlive?: boolean;
+}
+
+interface DefaultExport {
+    useSession: typeof useSession;
+    getSession: typeof session;
+    session: typeof session;
+    getProviders: typeof providers;
+    providers: typeof providers;
+    getCsrfToken: typeof csrfToken;
+    csrfToken: typeof csrfToken;
+    signin: typeof signin;
+    signIn: typeof signin;
+    signout: typeof signout;
+    signOut: typeof signout;
+    options: typeof setOptions;
+    setOptions: typeof setOptions;
+    Provider: typeof Provider;
+}
+
+type ContextProvider = FC<ContextProviderProps>;
+
+declare function useSession(): [Session, boolean];
+declare function session(context: NextPageContext): Promise<Session | null>;
+declare function providers(context: NextPageContext): Promise<GetProvidersResponse | null>;
+declare function csrfToken(context: NextPageContext): Promise<string | null>;
+declare function signin(provider: SessionProvider, data: GenericObject): Promise<void>;
+declare function signout(context: NextPageContext): Promise<void>;
+declare function setOptions(options: SetOptionsParams): void;
+declare const Provider: ContextProvider;
+
+declare const _default: DefaultExport;
+export default _default;
```

Comment:
Per https://github.com/DefinitelyTyped/DefinitelyTyped/pull/45775/files/ef8af8c4ce9c2ce40dcced2e35dcb61ff909e941#r466094897, these should be exported via `export { useSession, ... }` as was provided previously, rather than via `export default`.

Reply:
@rbuckton addressed 👍🏽